lib/_http_outgoing.js

```diff
@@ -138,15 +138,15 @@ OutgoingMessage.prototype._writeRaw = function(data, encoding, callback) {
       this.connection.writable &&
       !this.connection.destroyed) {
     // There might be pending data in the this.output buffer.
-    while (this.output.length) {
-      if (!this.connection.writable) {
-        this._buffer(data, encoding, callback);
-        return false;
+    var outputLength = this.output.length;
+    if (outputLength) {
+      for (var i = 0; i < outputLength; i++) {
+        var cb = this.outputCallbacks[i];
+        this.connection.write(this.output[i], this.outputEncodings[i], cb);
       }
-      var c = this.output.shift();
-      var e = this.outputEncodings.shift();
-      var cb = this.outputCallbacks.shift();
-      this.connection.write(c, e, cb);
+      this.output = [];
```

Comment:
the previous version didn't create new arrays. you can just do `this.output.length = 0;`. the same for the rest of these.